### ISSUE TITLE
Set specific hidden_size for ClassificationHead

### ISSUE THREAD

**Issue by GuillaumeDesforges** [huggingface/transformers#3048]
# 🚀 Feature request

Add a config term `head_hidden_size` to the model configurations that will be used for the head of models such as `RobertaForSequenceClassification`

## Motivation

HuggingFace transformers library provides a very accessible API and PyTorch models that can be used "plug n play" for various task such as classification.
In many case, varying the hidden size of the last layer (that outputs the logits) is one of the first thing we tweak to improve the performance on such a task.

Currently, the dense layer uses the `hidden_size` config parameter, which is the same as the one used in the transformer (BERT). One cannot change the hidden size of the last layer without changing the hidden size of the entire transformer model behind it.

This means we have to code a new PyTorch module in order to do something as simple as that.

## Your contribution

I could PR this should the change be welcomed.
